src/scaffoldfitter/fitterstepalign.py

```diff
@@ -71,28 +106,40 @@ def setRotation(self, rotation):
         0 = azimuth (about z)
         1 = elevation (about rotated y)
         2 = roll (about rotated x)
+        :return: True if state changed, otherwise False.
         """
         assert len(rotation) == 3, "FitterStepAlign:  Invalid rotation"
-        self._rotation = rotation
+        if rotation != self._rotation:
+            self._rotation = copy.copy(rotation)
+            return True
+        return False
 
     def getScale(self):
         return self._scale
 
     def setScale(self, scale):
         """
         :param scale: Real scale.
+        :return: True if state changed, otherwise False.
         """
-        self._scale = scale
+        if scale != self._scale:
+            self._scale = scale
```

Comment:
Wouldn't we need `self._scale = copy.copy(scale)` here too?

Reply:
Not in this case because it is a scalar.